### ISSUE TITLE
Time interval information

### ISSUE THREAD

**Issue by empyriumz** [v1xerunt/StageNet#6]
Hi there,

I find your paper quite interesting and it's great that you open sourced the code.
I'm trying to follow your code and reproducing the experiment results using the MIMIC-III dataset.
I notice that in the train.py:
`batch_time = torch.ones((batch_x.size(0), batch_x.size(1)), dtype=torch.float32).to(device)`
is used as input for the time interval between different visits, which is in contrast with what claimed in the paper.

Could you explain what is going on here? Is there something that I'm misunderstanding?
Thanks!

**Comment by v1xerunt** [comment]
Hi, it's the time interval between visits. For mimic-iii dataset, the biomarkers are sampled at each hour, so each timestep is actually one hour. So the time interval between each timestep (visit) is regular, and we just use the one-matrix. For another ESRD dataset used in our paper, the time interval between each visit is quite irregular (5-30days). In that case, the batch_time matrix is the real interval matrix.

**Comment by empyriumz** [comment]
Thanks for answering my questions!
Now I understand the reason why `batch_time` variable is uniform in this case.

However, I find that another variable `batch_interval` actually keeps track of the time interval for each physiologic variables, and is subsequently combined with `batch_x`:

`batch_x = torch.cat((batch_x, batch_interval), dim=-1)`

as the input, together with `batch_time`, to the neural network. 
So my questions are:

1. What is the variable representing the time interval sequence T=[\Delta_1...\Delta_t] mentioned in the paper?
2. Before concatenating with `batch_interval`,  `batch_x` contains 59 dimensional actual physiologic data (after one-hot encoding), plus a 17-dimensional mask. What is the purpose of putting the mask as part of the input data? 

I'm looking forward to your answer, thanks.

**Comment by v1xerunt** [comment]
batch_interval is not a required input of our model. One of our baselines DecayLSTM uses it to calculate the decay rate. For a fair comparison, we also input this to all other models. The time interval sequence used in our paper is batch_time.

The 17-dimensional mask is the original input in the benchmark code. We do not make any change in this part.

**Comment by empyriumz** [comment]
I see, thanks for your explanation!
